java/plasma/src/main/java/org/apache/arrow/plasma/ObjectStoreLink.java

```diff
@@ -123,4 +123,9 @@ void put(byte[] objectId, byte[] value, byte[] metadata)
    * @param objectId used to identify an object.
    */
   boolean contains(byte[] objectId);
+
+  /**
+   * List all objects in the PlasmaStore.
+   */
+  List<byte[]> list();
```

Comment:
I would suggest to chance the method signature to ```int list(List<byte[]> entries)``` so that users can get return code understanding whether it's going well for the method call.

Reply:
Can you clarify what you mean by this, if things aren't going well I'd expect an exception to be thrown?

Reply:
Either (throw an exception or via return code) way should be fine to make sure incorrect cases are handled well. If by exception, native exception should be catch (e.g. around L249) and throw via Java exception.

Reply:
modified by throw an exception.